packages/suite/src/reducers/suite/messageSystemReducer.ts

```diff
@@ -100,47 +100,43 @@ const messageSystemReducer = (
 const comparePriority = (a: Message, b: Message) => b.priority - a.priority;
 
 const makeSelectActiveMessagesByCategory = (category: Category) =>
-    createSelector(
-        (state: MessageSystemRootState) => state.messageSystem.config,
-        (state: MessageSystemRootState) => state.messageSystem.validMessages,
-        (state: MessageSystemRootState) => state.messageSystem.dismissedMessages,
-        (config, validMessages, dismissedMessages) => {
-            const nonDismissedValidMessages = validMessages[category].filter(
-                id => !dismissedMessages[id]?.[category],
-            );
+    memoize((state: MessageSystemRootState) => {
+        const { config, validMessages, dismissedMessages } = state.messageSystem;
+        const nonDismissedValidMessages = validMessages[category].filter(
+            id => !dismissedMessages[id]?.[category],
+        );
 
-            const messages = config?.actions
-                .filter(({ message }) => nonDismissedValidMessages.includes(message.id))
-                .map(action => action.message);
+        const messages = config?.actions
+            .filter(({ message }) => nonDismissedValidMessages.includes(message.id))
+            .map(action => action.message);
 
-            if (!messages?.length) return [];
+        if (!messages?.length) return [];
 
-            return messages.sort(comparePriority);
-        },
-    );
+        return messages.sort(comparePriority);
+    });
 
 export const selectActiveBannerMessages = makeSelectActiveMessagesByCategory('banner');
 export const selectActiveContextMessages = makeSelectActiveMessagesByCategory('context');
 export const selectActiveModalMessages = makeSelectActiveMessagesByCategory('modal');
 export const selectActiveFeatureMessages = makeSelectActiveMessagesByCategory('feature');
 
-export const selectBannerMessage = createSelector(
-    selectActiveBannerMessages,
-    messages => messages[0],
-);
+export const selectBannerMessage = memoize((state: MessageSystemRootState) => {
+    const activeBannerMessages = selectActiveBannerMessages(state);
+    return activeBannerMessages[0];
+});
 
-export const selectContextMessage = createSelector(
-    selectActiveContextMessages,
-    (_state: MessageSystemRootState, domain: ContextDomain) => domain,
-    (messages, domain) => messages.find(message => message.context?.domain === domain),
+export const selectContextMessage = memoizeWithArgs(
```

Comment:
Is it always better to use `memoizeWithArgs` and not just `memoize` and wrapping the arguments in object like
```javascript
memoize(({state, domain}: SelectContextMessageProps) => ...
```

Reply:
Is this question or statement?

Because proposed usage will completely break all memoization because you will need do this:

```
const something = useSelector(state => selectSomething({ state, domain })
```

And it will create that wrapping object with every rerender so it will rerender all the time. Only solution to this is wrap selector in another `useCallback` or `useMemo` which is extra work.

I already had similar problem with these rerenders in `suite-native/accounts/src/components/AccountsListGroup.tsx`

Reply:
👍 thanks for the explanation